### DISCUSSION TITLE
Could I create image from buffer or USM device pointer?

### DISCUSSION THREAD

**Discussion by CaoZhongZ** [intel/llvm#3178] [Q&A] [answered]
Can I do that? Without moving data between host and device.

**Comment by AlexeySachkov** [comment]
Hi @CaoZhongZ, looking at both [SYCL 1.2.1](https://www.khronos.org/registry/SYCL/specs/sycl-1.2.1.pdf) and [SYCL 2020 Provisional](https://www.khronos.org/registry/SYCL/specs/sycl-2020-provisional.pdf) I don't see a way to do so.

I see that in [OpenCL](https://www.khronos.org/registry/OpenCL/sdk/2.2/docs/man/html/clCreateImage.html) there is an image type `CL_​MEM_​OBJECT_​IMAGE1D_​BUFFER`, which means that an image is created from another image of from a buffer, but it still a host API and it accepts a host memory object as its input.

**Comment by jbrodman** [comment, marked as answer]
Creating an image from a USM pointer is not currently supported.
